### PR TITLE
Import collection

### PR DESCRIPTION
**Pull request by imabdulbasit**

### CODE REVIEW

File: consumer/src/events.rs [hunk 1]
```diff
@@ -113,22 +125,28 @@ impl EventKind {
...
+            EventKind::MintDrop => Ok(SolanaNftEvent::MintDropSigningRequested(tx)),
+            EventKind::UpdateDrop => Ok(SolanaNftEvent::UpdateDropSigningRequested(tx)),
+            EventKind::TransferAsset => Ok(SolanaNftEvent::TransferAssetSigningRequested(tx)),
+            EventKind::RetryCreateDrop => Ok(SolanaNftEvent::RetryCreateDropSigningRequested(tx)),
+            EventKind::RetryMintDrop => Ok(SolanaNftEvent::RetryMintDropSigningRequested(tx)),
+            EventKind::ImportCollection => Err(ProcessorError::new(
+                ProcessorErrorKind::Solana(anyhow!("Invalid Operation")),
+                EventKind::ImportCollection,
+                ErrorSource::ImportCollectionFailure,
+            )),
```
Comment (kespinola) [inline consumer/src/events.rs:149]: Hmm. Not a transaction. Can we not use the same path as the transaction types?

File: consumer/src/events.rs [hunk 2]
```diff
@@ -677,4 +714,294 @@ impl Processor {
...
+        .iter()
+        .map(|c| SolanaCreator {
+            address: c.address.to_string(),
+            verified: c.verified,
+            share: c.share,
+        })
+        .collect::<Vec<_>>();
+
+    #[allow(clippy::cast_sign_loss)]
+    let uuid = Uuid::from_u64_pair(Utc::now().timestamp_nanos() as u64, rand::random::<u64>());
```
Comment (kespinola) [inline consumer/src/events.rs:944]: Since addresses are unique why not use it as it as part of the pair?

File: consumer/src/events.rs [hunk 3]
```diff
@@ -518,14 +555,14 @@ impl Processor {
...
+        let collection = collections::ActiveModel {
+            id: Set(id),
+            master_edition: Set(master_edition.to_string()),
+            owner: Set(owner.to_string()),
+            metadata: Set(metadata.to_string()),
+            associated_token_account: Set(associated_token_account.to_string()),
+            mint: Set(mint.to_string()),
+            update_authority: Set(update_authority.to_string()),
             ..Default::default()
         };
```
Comment (kespinola) [inline consumer/src/events.rs:567]: The create function converts the model to an active model.

**Review by kespinola — CHANGES_REQUESTED** [review]

File: consumer/src/events.rs (hunk 3 above)
Comment (imabdulbasit) [reply consumer/src/events.rs:567]: I  was using default generated id before. 
FYI Default values for model and active model are different. Example, default id will be zero uuid for model but active model is going to set at db side.

File: consumer/src/events.rs (hunk 2 above)
Comment (imabdulbasit) [reply consumer/src/events.rs]: Address has 32 bytes so can only use first 8 bytes for the other half or 16 bytes for both parts so I think random number would be a better option?

**Review by kespinola — APPROVED** [review]

**Pull request merged**